### ISSUE TITLE
mellon_create_metadata.sh never generates metadata files with /dev/urandom

### ISSUE THREAD

**Issue by gaal-dev** [latchset/mod_auth_mellon#118]
Hi. I added "set -x" to this script to see what happens.

`
...
+ umask 0077
++ mktemp -t mellon_create_sp.XXXXXXXXXX
+ TEMPLATEFILE=/tmp/mellon_create_sp.rWskeaitUC
+ cat
+ openssl req -utf8 -batch -config /tmp/mellon_create_sp.rWskeaitUC -new -x509 -days 3652 -nodes -out http_apache_mellon_metadata.cert -keyout http_apache_mellon_metadata.key`

The script and the last command were executed in a Docker container.

`root@apache:~# openssl req -utf8 -batch -config /tmp/mellon_create_sp.rWskeaitUC -new -x509 -days 3652 -nodes -out http_apache_mellon_metadata.cert -keyout http_apache_mellon_metadata.key
...
Cannot write random bytes:
4097216A697F0000:error:1200007A:random number generator:RAND_write_file:Not a regular file:../crypto/rand/randfile.c:190:Filename=/dev/urandom`

In fact, this code has never been executed. 

`
...
rm -f "$TEMPLATEFILE" "${TEMPLATEFILE}.RANDOM"

CERT="$(grep -v '^-----' "$OUTFILE.cert")"

cat >"$OUTFILE.xml" <<EOF`

The execution stops because the script contains the instruction 'set -e', but redirecting stderr to /dev/null hides that error message and the error reason is not obvious.

**Comment by gaal-dev** [comment]
> Were you in fact using mellon 0.18.1?

I run the script in a Docker container and could not see why metadata files were not created becuase it happened silently

`# apt info libapache2-mod-auth-mellon
Package: libapache2-mod-auth-mellon
Version: 0.18.0-1build1
Priority: extra
Section: web
Origin: Ubuntu
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>`

**Comment by thijskh** [comment]
Yes, so 0.18.1 should at least have fixed why they weren't created. But agreed that we could improve the error reporting of the script to not be silent.
